src/R/Editor/Impl/QuickInfo/QuickInfoSource.cs

```diff
@@ -57,61 +59,82 @@ internal sealed class QuickInfoSource : IQuickInfoSource {
+            // on the function arguments.
+            string functionName = GetFunctionName(ast, position);
             if (!string.IsNullOrEmpty(functionName)) {
-                ITextSnapshot snapshot = session.TextView.TextBuffer.CurrentSnapshot;
-
-                position = Math.Min(signatureEnd, position);
-                int start = Math.Min(position, snapshot.Length);
-                int end = Math.Min(signatureEnd, snapshot.Length);
-
-                applicableToSpan = snapshot.CreateTrackingSpan(Span.FromBounds(start, end), SpanTrackingMode.EdgeInclusive);
-                packageName = packageName ?? _packageName;
-                _packageName = null;
-
-                var functionInfo = FunctionIndex.GetFunctionInfo(functionName, packageName, retriggerAction, session);
-
-                if (functionInfo != null && functionInfo.Signatures != null) {
-                    foreach (ISignatureInfo sig in functionInfo.Signatures) {
-                        string signatureString = sig.GetSignatureString(functionName);
-                        int wrapLength = Math.Min(SignatureInfo.MaxSignatureLength, signatureString.Length);
-                        string text;
-
-                        if (string.IsNullOrWhiteSpace(functionInfo.Description)) {
-                            text = string.Empty;
-                        } else {
-                            /// VS may end showing very long tooltip so we need to keep 
-                            /// description reasonably short: typically about
-                            /// same length as the function signature.
-                            text = signatureString + "\r\n" + functionInfo.Description.Wrap(wrapLength);
-                        }
-
-                        if (text.Length > 0) {
-                            quickInfoContent.Add(text);
-                            return true;
-                        }
-                    }
-                }
+                functionInfo = FunctionIndex.GetFunctionInfo(functionName, packageName, retriggerAction, session);
             }
 
+            if (functionInfo?.Signatures == null) {
+                return false;
+            }
+
+            foreach (var sig in functionInfo.Signatures) {
+                string signatureString = sig.GetSignatureString(functionName);
+                int wrapLength = Math.Min(SignatureInfo.MaxSignatureLength, signatureString.Length);
+                string text;
+
+                if (string.IsNullOrWhiteSpace(functionInfo.Description)) {
+                    text = string.Empty;
+                } else {
+                    // VS may end showing very long tooltip so we need to keep 
+                    // description reasonably short: typically about
+                    // same length as the function signature.
+                    text = signatureString + "\r\n" + functionInfo.Description.Wrap(wrapLength);
```

Comment:
Yeah, I will look into cleaning this up in the editor split. There is bunch of old code here.